e3nn/point/data_helpers.py

```diff
@@ -197,6 +209,13 @@ def _neighbor_list_and_relative_vec(pos, r_max, self_interaction=True, cell=None
         torch.LongTensor(first_idex),
         torch.LongTensor(second_idex)
     ))
+    # We have to compute displacements in pytorch so that they are backwardable
+    displacements = pos[second_idex] - pos[first_idex]
+    displacements = displacements + torch.einsum(
+        'ni,ij->nj',
+        torch.as_tensor(shifts, dtype=displacements.dtype),
+        torch.as_tensor(cell, dtype=displacements.dtype)
+    )
     edge_attr = torch.as_tensor(displacements)
```

Comment:
Almost, good catch — it is necessary but needed to be earlier to deal gracefully with numpy array inputs. (Otherwise would get `TypeError` from sum of ndarray with Tensor.) Fixed in last commit.